core/src/mindustry/editor/MapEditorDialog.java

```diff
@@ -691,28 +707,32 @@ private void addBlockSelection(Table table){
             return Integer.compare(b1.id, b2.id);
         });
 
+        int i = 0;
+
         for(Block block : blocksOut){
             TextureRegion region = block.icon(Cicon.medium);
 
-            if(!Core.atlas.isFound(region) || !block.inEditor || (block.buildVisibility == BuildVisibility.debugOnly)) continue;
+            if(!Core.atlas.isFound(region) || !block.inEditor
+                    || block.buildVisibility == BuildVisibility.debugOnly
+                    || (!block.localizedName.toLowerCase().contains(searchText.toLowerCase()) && !searchText.isEmpty())
```

Comment:
`!searchText.isEmpty()` should be put first so it short-circuits and prevents unnecessary `contains()` checks.

Reply:
Done.